system/BaseModel.php

```diff
@@ -1656,8 +1656,10 @@ protected function objectToArray($data, bool $onlyChanged = true, bool $recursiv
     {
         $properties = $this->objectToRawArray($data, $onlyChanged, $recursive);
 
+        assert(is_array($properties));
+
         // Convert any Time instances to appropriate $dateFormat
-        if ($properties) {
+        if ($properties !== []) {
```

Comment:
@kenjis I prefer `$properties !== null` check for this, so it will always be array, `array_map()` on empty array will less or no different on usage I think.

```suggestion
        // Convert any Time instances to appropriate $dateFormat
        if ($properties !== null) {
```

Reply:
oh, I see, it always be an array on 4.5 as you said :)